app/src/main/java/arjun/hanabiandroid/game/Move.java

```diff
@@ -0,0 +1,36 @@
+package arjun.hanabiandroid.game;
+
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class Move {
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({
+            MOVE_TYPE_PLAY,
+            MOVE_TYPE_DISCARD,
+            MOVE_TYPE_HINT,
+    })
+    @interface MoveType {
+    }
+
+    public static final int MOVE_TYPE_PLAY = 0;
+    public static final int MOVE_TYPE_DISCARD = 1;
+    public static final int MOVE_TYPE_HINT = 2;
+
+    @MoveType
+    public final int moveType;
+    public final int cardIndex;     // Only used for play and discard
+    public final int playerIndex;   // Only used for hint moves
+    public final int hintedRank;    // Only used for hint moves
+    public final int hintedSuit;    // Only used for hint moves
```

Comment:
I did this for a few reasons:
1. Move is so far just a model class, i.e. only holds data, and isn't responsible for any logic. It is confusing to hold data in subclasses, since the controller responsible for using this data might then have to cast to subclasses
2. It will be easier to serialize this class to a database, since subclassing is a nightmare for that.